### ISSUE TITLE
Define the way PreAction and Test template will behave

### ISSUE THREAD

**Issue by filfreire** [tranquility-bdd/tranquility-bdd#26]
We need to define how PreAction and Test templates will act, taking into account they can:
- get code that will be executed only during `run` phase of PreAction/Test
- that code can change Environment variables - specially setup new ones, or update existing ones - "seamlessly".

The outcome of this github issue will be another issue where we implement the decided approach for PreAction and Test templates.

/cc @jmromanos

**Comment by jmromanos** [comment]
PreAction in my action should be a function defined by the user that works on an Action. Following this approach Test should be also a function but that works on a Response.

A PreAction, Action and Test should form a Request. The Request should have a Run method that would do the following
```
Action.PreAction()
res, err := Action.Run()
if err != nil {
   panic(err)
}
res.Test()
```

Then we could have a Runner that could execute all Requests given by the user.

**Comment by filfreire** [comment]
@jmromanos looks like a good initial approach.

And maybe for now, since we're more focused on REST testing, we can mimic Postman's naming:

- `Request` - PreAction + Action + Test
- `Folder` - Set of Sequential Requests - which own `Run` method calls the Request's `Run`
- `Collection` - Set of Folders - similar idea to above (with a possible nice-to-have: ability of running Folders in parallel, and each Folder works from the start it's own clone copy of the `Env` )

But, we can also change the names a bit. It's up to us 😄 

If you think of a better generic name for the `Request` feel free.

**Comment by filfreire** [comment]
@jmromanos we can close this one since it's done.
